src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -317,5 +317,27 @@ private string FormatEnumValueToString(string value, JavaScriptEncoder? encoder)
 
             return converted;
         }
+
+        internal override T ReadWithQuotes(ReadOnlySpan<byte> utf8Bytes)
+        {
+            string enumString = JsonReaderHelper.TranscodeHelper(utf8Bytes);
+
+            if (!Enum.TryParse(enumString, out T value)
+                && !Enum.TryParse(enumString, ignoreCase: true, out value))
+            {
+                ThrowHelper.ThrowJsonException();
+                return default;
+            }
+
+            return value;
+        }
+
+        internal override void WriteWithQuotes(Utf8JsonWriter writer, T value, JsonSerializerOptions options, ref WriteStack state)
+        {
+            string enumString = value.ToString();
+            writer.WritePropertyName(enumString);
+        }
+
+        internal override bool CanBeDictionaryKey => true;
```

Comment:
Right, I will add logic to honor the enum policy, thanks.

Reply:
Keep in mind that we are accounting only for the policy, the `EnumConverterOptions` that defines whether we allow strings and/or numbers is not being considered since all property names are strings.

Reply:
Actually, we should not account for a naming policy. Any `EnumConverter` that we use for keys can only be created by internal code so no policy is provided.